paf-mvp-demo-express/src/index.ts

```diff
@@ -68,6 +74,17 @@ const addMiddleware = (app: Express) => {
   app.use((req, res, next) => {
     req.secure ? next() : res.redirect(`https://${req.headers.host}${req.url}`);
   });
+
+  // Set the CORS access control allow origin if a proxy path and the origin is provided in the request.
+  // This is needed to support pages that are loaded form the file system to quickly demo the functionality. There
+  // may be better places to place this code.
+  app.use((req, res, next) => {
+    if (req.path.startsWith('/paf-proxy/')) {
```

Comment:
⚠️ Big mistake!
The proxy relies heavily on the `origin` header, and partially on CORS. This piece of code invalidates it entirely.
If you need some specific shortcuts for accessing "file system" files, please:
- give the rationale behind it (I didn't get exactly how your "example HTML" file is used)
- if specific code is needed for this file, restrict the changes **to this particular file only**.
- if this is for the local development, restrict it to the `isLocalDev` condition
- don't put it in the "middleware" (which will be applied to all webapps, APIs or web sites) but only specifically in the website declarations (ex: `paf-market.ts`)

Reply:
Developers like to get up and running quickly. The example.html can be opened from the file system and shows the solution running in their environment. The change adds the origin header in the response if present. Otherwise the wildcard is used.
Given the approach being taken the origin validation needs to be somewhere. In this case it only applies if paf-proxy is present. Can you point me to where the origin validation is?
The alternative is to remove example.html and require the developer to get the MVP running in their environment before they can play with a publisher page.

Reply:
To sum up, I don't see how we can make an "example HTML" page be considered valid to call the PAF client (aka proxy) and operator, without introducing dangerous hacks in the code.
Plus, even if I get the point to make developers' tests as smooth as possible, I don't really see the difference between accessing https://www.pifdemopublisher.com (running locally) versus opening the example.html page?

Reply:
The example.html and associated changes to index.ts have been removed in the latest commit. As you say it may be too complex to consider in the MVP at this stage. Individual CMP providers can address this in their solutions if they wish.